src/bdkad/KademliaHandler.cpp

```diff
@@ -326,4 +364,57 @@ namespace bdkad
       context.writeError("Failed", "Value not found", bdhttp::ErrorCode::GENERIC_ERROR);
     }
   }
+
+  void KademliaHandler::OnGetActivityLog(bdhttp::HttpContext & context)
+  {
+    printf("Kademlia::OnGetActivityLog\n");
+
+    uint32_t starttime = static_cast<uint32_t>(strtoul(context.parameter("starttime"), nullptr, 10));
+    uint32_t endtime   = static_cast<uint32_t>(strtoul(context.parameter("endtime"), nullptr, 10));
+
+
+    std::string query("type:\"log:storage\"");
```

Comment:
Use flooding to query logs is wrong. Query for hosts will usually work because the client needs only a few result (e.g. 8 hosts). So the flooding algorithm should work because the stopping criteria should be hit pretty soon (or timed out within a particular period). Query for logs is different since there is no ending criteria until all kademlia nodes have been reached.

What we discussed is to have a single KEY to store all the logs. For this new LOG opcode, when the node who is supposed to store the data receives a new log, it does not replace the old stored data like "SetValue" but instead "append" the data (may require deduplicate though - do not store the same log twice).